src/site/twitch.tv/modules/media-player/MediaPlayer.vue

```diff
@@ -0,0 +1,155 @@
+import { useTooltip } from "@/composable/useTooltip";
+import ForwardIcon from "@/assets/svg/icons/ForwardIcon.vue";
+import GaugeIcon from "@/assets/svg/icons/GaugeIcon.vue";
+import VideoStatsTooltip from "./VideoStatsTooltip.vue";
+
+const props = defineProps<{
+	instance: HookedInstance<Twitch.MediaPlayerComponent>;
+	videoStatsTeleportLocation: HTMLElement | null | undefined;
+}>();
+
+const teleportLocation = toRef(props, "videoStatsTeleportLocation");
+const doTeleport = ref<boolean>(false);
+
+const shouldShowVideoStats = useConfig<boolean>("player.video_stats");
+const actionOnClick = useConfig<number>("player.action_onclick");
+
+const liveLatency = ref<string>();
+const videoStats = ref({
+	droppedFrames: 0,
+	playbackRate: 0,
+	bitrate: "0",
+	width: 0,
+	height: 0,
+	framerate: 0,
+});
+
+definePropertyHook(props.instance.component, "props", {
+	value: (v: Twitch.MediaPlayerComponent["props"]) => {
+		if (v.mediaPlayerInstance) {
+			definePropertyHook(v.mediaPlayerInstance.core.state, "liveLatency", {
+				value: (v: number) => {
+					liveLatency.value = v?.toFixed(2) ?? "-.--";
+				},
+			});
+		}
+
+		// we need to re-add the event listener when the mediaPlayer props changes
+		addClickHandler();
+	},
+});
+
+watch(
+	liveLatency,
+	() => {
+		const mediaPlayer = props.instance.component.props?.mediaPlayerInstance;
+		if (mediaPlayer) {
+			videoStats.value.droppedFrames = mediaPlayer.getDroppedFrames();
+			videoStats.value.bitrate = (mediaPlayer.getVideoBitRate() / 1000)?.toFixed(0) ?? "0";
+			videoStats.value.framerate = mediaPlayer.getVideoFrameRate();
+			videoStats.value.playbackRate = mediaPlayer.getPlaybackRate();
+			videoStats.value.width = mediaPlayer.getVideoWidth();
+			videoStats.value.height = mediaPlayer.getVideoHeight();
+		}
+	},
+	{ immediate: true },
+);
+
+const videoStatsRef = ref();
+const videoStatsTooltip = useTooltip(VideoStatsTooltip, videoStats.value);
+
+watch(
+	teleportLocation,
+	(n) => {
+		n ? (doTeleport.value = true) : (doTeleport.value = false);
+	},
+	{ immediate: true },
+);
+
+function pauseOnClick() {
+	if (!props.instance.component.props) return;
+	const mediaPlayer = props.instance.component.props.mediaPlayerInstance;
+	if (mediaPlayer.isPaused()) {
+		mediaPlayer.play();
+	} else {
+		mediaPlayer.pause();
+	}
+}
+
+function muteOnClick() {
+	if (!props.instance.component.props) return;
+	const mediaPlayer = props.instance.component.props.mediaPlayerInstance;
+	mediaPlayer.setMuted(!mediaPlayer.isMuted());
+}
+
+function addClickHandler() {
+	const mediaPlayer = props.instance.component.props?.mediaPlayerInstance;
+	if (!mediaPlayer) return;
+	// getHTMLVideoELement function not working as expected
+	const videoElement = mediaPlayer.core.mediaSinkManager.video;
+	if (!videoElement) return;
+
+	const overlayContainer = videoElement.nextElementSibling?.querySelector<HTMLElement>(
+		"div[data-a-target='player-overlay-click-handler']",
+	);
```

Comment:
nit: can avoid optional chaining

```suggestion
	const videoElement = mediaPlayer.core.mediaSinkManager.video;
	if (!videoElement || !videoElement.nextElementSibling) return;

	const overlayContainer = videoElement.nextElementSibling.querySelector<HTMLElement>(
		"div[data-a-target='player-overlay-click-handler']",
	);
```